__tests__/app.test.js

```diff
@@ -0,0 +1,112 @@
+const request = require("supertest");
+const app = require("../app");
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed");
+const testData = require("../db/data/test-data");
+
+beforeEach(() => seed(testData));
+afterAll(() => db.end());
+
+describe("GET /api/healthcheck", () => {
+  test("it should respond with a 200 status code", () => {
+    return request(app).get("/api/topics").expect(200);
+  });
+});
+
+describe("GET /api/topics", () => {
+  test("200: responds with an array of topic objects", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.topics).toBeInstanceOf(Array);
+        body.topics.forEach((topic) => {
+          expect(topic).toEqual(
+            expect.objectContaining({
+              slug: expect.any(String),
+              description: expect.any(String),
+            })
+          );
+        });
+      });
+  });
+
+  test("404: responds with a 404 for an invalid endpoint", () => {
+    return request(app)
+      .get("/api/not-a-route")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Route not found");
+      });
+  });
+});
+
+describe("GET /api", () => {
+  it("should respond with a JSON object describing all available endpoints", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((response) => {
+        const endpoints = response.body;
+        expect(endpoints).toHaveProperty("GET /api");
+        expect(endpoints["GET /api"]).toEqual({
+          description:
+            "serves up a json representation of all the available endpoints of the api",
+        });
+        expect(endpoints).toHaveProperty("GET /api/topics");
+        expect(endpoints["GET /api/topics"]).toEqual({
+          description: "serves an array of all topics",
+          queries: [],
+          exampleResponse: {
+            topics: [
+              {
+                slug: "football",
+                description: "Footie!",
+              },
+            ],
+          },
+        });
+        expect(endpoints).toHaveProperty("GET /api/articles/:article_id");
+        expect(endpoints["GET /api/articles/:article_id"]).toEqual({
```

Comment:
Rather than having to do all of these expect statements and checking how each endpoint looks, we can simplify it by just requiring in the endpoints.json file, then expecting our response to equal it. This means it'll be flexible as the file changes and we won't need to keep changing the test. 👍